app/src/main/java/com/example/weatherandroidapp/viewmodels/WeatherGridViewModel.kt

```diff
@@ -1,15 +1,34 @@
 package com.example.weatherandroidapp.viewmodels
 
-import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.asLiveData
+import androidx.lifecycle.viewModelScope
 import com.example.domain.entities.ResponseWeatherInfo
 import com.example.domain.usecase.WeatherUseCaseInterface
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 
 class WeatherGridViewModel(
     private val useCaseWeatherCity: WeatherUseCaseInterface
 ) : ViewModel() {
-    private val weatherInfoLiveData: LiveData<ResponseWeatherInfo> = useCaseWeatherCity.observeWeather().asLiveData()
+    private val _weatherInfoLiveData = MutableLiveData<ResponseWeatherInfo>()
```

Comment:
this LD with the observer its not observed. you can delete it

Reply:
Done